grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -229,7 +243,7 @@ V addInstance(Schema.BaseType instanceBaseType, BiFunction<Vertex, T, V> produce
     public void delete(){
         checkTypeMutation();
         boolean hasSubs = getVertex().edges(Direction.IN, Schema.EdgeLabel.SUB.getLabel()).hasNext();
-        boolean hasInstances = getVertex().edges(Direction.IN, Schema.EdgeLabel.ISA.getLabel()).hasNext();
+        boolean hasInstances = ((TypeImpl) currentShard()).getIncomingNeighbours(Schema.EdgeLabel.ISA).findAny().isPresent();
```

Comment:
This will allow the type to be deleted even if it has another shard which still has instances.